lib/AST/GenericSignatureBuilder.cpp

```diff
@@ -4102,6 +4102,11 @@ ConstraintResult GenericSignatureBuilder::expandConformanceRequirement(
         auto type = dyn_cast<TypeDecl>(found);
         if (!type || isa<AssociatedTypeDecl>(type)) continue;
 
+        // Ignore nominal types defined in this protocol. They're always
+        // invalid declarations.
+        if (isa<NominalTypeDecl>(type) && type->getDeclContext() == proto)
```

Comment:
The right hand side of the && is not necessary. We should be filtering out extension members as well